include/xtensor/xmath.hpp

```diff
@@ -1975,6 +1978,35 @@ XTENSOR_INT_SPECIALIZATION_IMPL(FUNC_NAME, RETURN_VAL, unsigned long long);
         using result_type = big_promote_type_t<typename std::decay_t<E>::value_type>;
         return accumulate(make_xaccumulator_functor(detail::nan_multiplies<result_type>(), detail::nan_init<result_type, 1>()), std::forward<E>(e));
     }
+
+    /**
+     * @ingroup red_functions
+     * @brief Integrate along the given axis using the composite trapezoidal rule.
+     *
+     * Returns definite integral as approximated by trapezoidal rule.
+     * @param y an \ref xexpression
+     * @param dx the spacing between sample points (optional)
+     * @param axis the axis along which to integrate.
+     * @return an xarray or xtensor depending on shape type
+     */
+    template <class T>
+    auto trapz(const xexpression<T>& y, typename T::value_type dx = 1.0, std::ptrdiff_t axis = -1)
+    {
+        auto& yd = y.derived_cast();
+        std::size_t saxis = static_cast<std::size_t>(axis);
+
+        if (axis == -1)
+            saxis = yd.dimension() - 1;
+
+        slice_vector slice1(yd.dimension(), all());
+        slice_vector slice2(yd.dimension(), all());
+        slice1[saxis] = range(1, xnone());
+        slice2[saxis] = range(xnone(), yd.shape()[saxis] - 1);
+
+        auto trap = dx * (strided_view(yd, slice1) + strided_view(yd, slice2)) / 2.0;
```

Comment:
`* 0.5`

Reply:
I guess that's a good idea, as this will probably not be optimized by the compiler